unused/unused.go

```diff
@@ -12,28 +13,53 @@ type UnusedField struct {
 	Name string
 }
 
-type byName []UnusedField
-
-func (a byName) Len() int           { return len(a) }
-func (a byName) Less(i, j int) bool { return a[i].Name < a[j].Name }
-func (a byName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-
-type unusedRegistry map[string]UnusedField
+type unusedRegistry map[string]output.Field
 
 func (r unusedRegistry) Record(field parser.SchemaField) {
-	r[field.Name] = UnusedField{Name: field.Name}
+	r[field.Name] = output.Field{
+		Field: field.Name,
+		Line:  field.Line,
+		// Not recording the file as that would be the same for all (the schema)
+	}
 }
 
-func GetUnusedFields(schema *ast.Schema, queriesPaths []string) ([]UnusedField, error) {
-	unusedFields := make(unusedRegistry)
+func GetUnusedFields(schemas []string, queryFiles []string, verbose bool) (output.Data, error) {
+	out := output.Data{}
 
-	deprecatedFields := parser.ParseDeprecatedFields(schema)
+	for _, schemaSource := range schemas {
+		schema, err := sources.LoadSchema(schemaSource)
+		if err != nil {
+			return nil, err
+		}
 
-	queries, err := sources.LoadQueries(schema, queriesPaths)
-	if err != nil {
-		return []UnusedField{}, err
+		if verbose {
+			fmt.Println("debug: Succesfully loaded schema from", schemaSource)
+		}
+
+		queries, err := sources.LoadQueries(schema, queryFiles)
+		if err != nil {
+			return nil, err
+		}
+
+		out[schemaSource] = []output.Field{}
+		unusedFields, err := getUnusedFields(schema, queries)
+		if err != nil {
+			return nil, err
+		}
+		for _, field := range unusedFields {
+			out.AppendField(schemaSource, field)
```

Comment:
Are we doing this in other places? just wondering if we could make it look a bit more like `append` so it takes a variadic argument and we can avoid the external loop

Reply:
Good suggestion. In this case though we need the loop as `unusedFields` is a map, not a slice so making `AppendField` variadic doesn't help.